framework/test/gtest/BundleContextTest.cpp

```diff
@@ -192,4 +192,115 @@ TEST(BundleContextTest, NoSegfaultWithServiceFactory)
     }
     thread.join();
 }
+
+template <typename ServiceT>
+bool
+verifyOrdering(std::vector<cppmicroservices::ServiceReference<ServiceT>>& refs)
+{
+    if (refs.size() > 0)
+    {
+        auto last = refs[0];
+        for (auto const& ref : refs)
+        {
+            if (!(any_cast<int>(ref.GetProperty(Constants::SERVICE_RANKING))
+                  <= any_cast<int>(last.GetProperty(Constants::SERVICE_RANKING))))
+            {
+                return false;
+            }
+
+            last = ref;
+        }
+    }
+    return true;
+}
+
+TEST(BundleContextTest, TestGetServiceReferenceOrderingBasic)
+{
+    cppmicroservices::Framework framework = cppmicroservices::FrameworkFactory().NewFramework();
+    framework.Start();
+
+    auto context = framework.GetBundleContext();
+
+    // Create Services
+    (void)context.RegisterService<bc_tests::TestService>(std::make_shared<bc_tests::TestService>(),
+                                                         ServiceProperties {
+                                                             {Constants::SERVICE_RANKING, 2}
+    });
+    (void)context.RegisterService<bc_tests::TestService>(std::make_shared<bc_tests::TestService>(),
+                                                         ServiceProperties {
+                                                             {Constants::SERVICE_RANKING, 0}
+    });
+    (void)context.RegisterService<bc_tests::TestService>(std::make_shared<bc_tests::TestService>(),
+                                                         ServiceProperties {
+                                                             {Constants::SERVICE_RANKING, 4}
+    });
+
+    (void)context.RegisterService<bc_tests::TestService>(std::make_shared<bc_tests::TestService>(),
+                                                         ServiceProperties {
+                                                             {Constants::SERVICE_RANKING, 1}
+    });
+
+    (void)context.RegisterService<bc_tests::TestService>(std::make_shared<bc_tests::TestService>(),
+                                                         ServiceProperties {
+                                                             {Constants::SERVICE_RANKING, 3}
+    });
+
+    auto refs = context.GetServiceReferences<bc_tests::TestService>();
+
+    ASSERT_TRUE(verifyOrdering(refs));
+    ASSERT_TRUE(refs.size() == 5);
+
+    framework.Stop();
+    framework.WaitForStop(std::chrono::milliseconds::zero());
+}
+
+TEST(BundleContextTest, TestGetServiceReferenceOrderingTwoKeys)
```

Comment:
this test should be written using [value parameterized tests](https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests)

Reply:
fixed